-1 
Tested at: bccbe82f2271ea304aa146ec19792b8c5a2a9894
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 64 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/EventFilter/L1TRawToDigi/src/implementations_stage2/GlobalExtBlkPacker.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/EventFilter/L1TRawToDigi/src/implementations_stage2/GlobalExtBlkUnpacker.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/EventFilter/L1TRawToDigi/src/implementations_stage2/JetPacker.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/EventFilter/L1TRawToDigi/src/implementations_stage2/JetUnpacker.cc 
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/EventFilter/L1TRawToDigi/src/implementations_stage2/L1TObjectCollections.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/DataFormats/L1TGlobal/src/GlobalAlgBlk.cc:27:27: error: unused variable 'maxPhysicsTriggers' [-Werror,-Wunused-const-variable]
const static unsigned int maxPhysicsTriggers = 512;  //TO DO Move this to a standard parameter area.
                          ^
1 error generated.
>> Compiling  /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_0_X_2016-03-04-2300/src/EventFilter/L1TRawToDigi/src/implementations_stage2/L1TStage2Layer2Constants.cc 
Selected class -> edm::Wrapper<BXVector<GlobalAlgBlk> > for ROOT: edm::Wrapper<BXVector<GlobalAlgBlk> >
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13607/11726/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/4c8d0489ed5d877f25eb71b53b7844a425ee79dc
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13607/11726/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13607/11726/git-merge-result

--- a/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerObjectMapRecord.cc
+++ b/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerObjectMapRecord.cc
@@ -47,9 +47,9 @@ const L1GlobalTriggerObjectMap* L1GlobalTriggerObjectMapRecord::getObjectMap(
 
     // no algoName found, return zero pointer!
     edm::LogError("L1GlobalTriggerObjectMapRecord")
-        << "\n\n  ERROR: The requested algorithm name = " << algoNameVal
-        << "\n  does not exists in the trigger menu."
-        << "\n  Returning zero pointer for getObjectMap\n\n" << std::endl;
+        << " ERROR: The requested algorithm name = " << algoNameVal
+        << " does not exist in the trigger menu."
+        << " Returning zero pointer for getObjectMap." << std::endl;
 
     return 0;
 
@@ -71,9 +71,9 @@ const L1GlobalTriggerObjectMap* L1GlobalTriggerObjectMapRecord::getObjectMap(
 
     // no algoBitNumberVal found, return zero pointer!
     edm::LogError("L1GlobalTriggerObjectMapRecord")
-        << "\n\n  ERROR: The requested algorithm with bit number = " << algoBitNumberVal
-        << "\n  does not exists in the trigger menu."
-        << "\n  Returning zero pointer for getObjectMap\n\n" << std::endl;
+        << " ERROR: The requested algorithm with bit number = " << algoBitNumberVal
+        << " does not exist in the trigger menu."
+        << " Returning zero pointer for getObjectMap." << std::endl;
 
     return 0;
     
@@ -97,10 +97,10 @@ const CombinationsInCond* L1GlobalTriggerObjectMapRecord::getCombinationsInCond(
 
     // no (algoName, condName) found, return zero pointer!
     edm::LogError("L1GlobalTriggerObjectMapRecord")
-    << "\n\n  ERROR: The requested \n    (algorithm name, condition name) = ("
+    << " ERROR: The requested (algorithm name, condition name) = ("
     << algoNameVal << ", " << condNameVal
-    << ") \n  does not exists in the trigger menu."
-    << "\n  Returning zero pointer for getCombinationsInCond\n\n"
+    << ") does not exists in the trigger menu."
+    << " Returning zero pointer for getCombinationsInCond."
     << std::endl;
 
     return 0;
@@ -123,10 +123,10 @@ const CombinationsInCond* L1GlobalTriggerObjectMapRecord::getCombinationsInCond(
 
     // no (algoBitNumber, condName) found, return zero pointer!
     edm::LogError("L1GlobalTriggerObjectMapRecord")
-    << "\n\n  ERROR: The requested \n    (algorithm bit number, condition name) = ("
+    << " ERROR: The requested (algorithm bit number, condition name) = ("
     << algoBitNumberVal << ", " << condNameVal
-    << ") \n  does not exists in the trigger menu."
-    << "\n  Returning zero pointer for getCombinationsInCond\n\n"
+    << ") does not exists in the trigger menu."
+    << " Returning zero pointer for getCombinationsInCond."
     << std::endl;
 
     return 0;
@@ -149,10 +149,10 @@ bool L1GlobalTriggerObjectMapRecord::getConditionResult(
 
     // no (algoName, condName) found, return false!
     edm::LogError("L1GlobalTriggerObjectMapRecord")
-    << "\n\n  ERROR: The requested \n    (algorithm name, condition name) = ("
+    << " ERROR: The requested (algorithm name, condition name) = ("
     << algoNameVal << ", " << condNameVal
-    << ") \n  does not exists in the trigger menu."
-    << "\n  Returning false for condition result! Unknown result, in fact!\n\n"
+    << ") does not exists in the trigger menu."
+    << " Returning false for condition result! Unknown result, in fact!"
     << std::endl;
 
     return false;
@@ -175,10 +175,10 @@ bool L1GlobalTriggerObjectMapRecord::getConditionResult(
 
     // no (algoBitNumber, condName) found, return false!
     edm::LogError("L1GlobalTriggerObjectMapRecord")
-    << "\n\n  ERROR: The requested \n    (algorithm bit number, condition name) = ("
+    << " ERROR: The requested (algorithm bit number, condition name) = ("
     << algoBitNumberVal << ", " << condNameVal
-    << ") \n  does not exists in the trigger menu."
-    << "\n  Returning false for condition result! Unknown result, in fact!\n\n"
+    << ") does not exists in the trigger menu."
+    << " Returning false for condition result! Unknown result, in fact!"
     << std::endl;
 
     return false;

--- a/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerObjectMapRecord.cc
+++ b/DataFormats/L1GlobalTrigger/src/L1GlobalTriggerObjectMapRecord.cc
@@ -99,7 +99,7 @@ const CombinationsInCond* L1GlobalTriggerObjectMapRecord::getCombinationsInCond(
     edm::LogError("L1GlobalTriggerObjectMapRecord")
     << " ERROR: The requested (algorithm name, condition name) = ("
     << algoNameVal << ", " << condNameVal
-    << ") does not exists in the trigger menu."
+    << ") does not exist in the trigger menu."
     << " Returning zero pointer for getCombinationsInCond."
     << std::endl;
 
@@ -125,7 +125,7 @@ const CombinationsInCond* L1GlobalTriggerObjectMapRecord::getCombinationsInCond(
     edm::LogError("L1GlobalTriggerObjectMapRecord")
     << " ERROR: The requested (algorithm bit number, condition name) = ("
     << algoBitNumberVal << ", " << condNameVal
-    << ") does not exists in the trigger menu."
+    << ") does not exist in the trigger menu."
     << " Returning zero pointer for getCombinationsInCond."
     << std::endl;
 
@@ -151,7 +151,7 @@ bool L1GlobalTriggerObjectMapRecord::getConditionResult(
     edm::LogError("L1GlobalTriggerObjectMapRecord")
     << " ERROR: The requested (algorithm name, condition name) = ("
     << algoNameVal << ", " << condNameVal
-    << ") does not exists in the trigger menu."
+    << ") does not exist in the trigger menu."
     << " Returning false for condition result! Unknown result, in fact!"
     << std::endl;
 
@@ -177,7 +177,7 @@ bool L1GlobalTriggerObjectMapRecord::getConditionResult(
     edm::LogError("L1GlobalTriggerObjectMapRecord")
     << " ERROR: The requested (algorithm bit number, condition name) = ("
     << algoBitNumberVal << ", " << condNameVal
-    << ") does not exists in the trigger menu."
+    << ") does not exist in the trigger menu."
     << " Returning false for condition result! Unknown result, in fact!"
     << std::endl;
 

--- a/DataFormats/L1TGlobal/src/GlobalAlgBlk.cc
+++ b/DataFormats/L1TGlobal/src/GlobalAlgBlk.cc
@@ -24,7 +24,6 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include "FWCore/MessageLogger/interface/MessageDrop.h"
 
-const static unsigned int maxPhysicsTriggers = 512;  //TO DO Move this to a standard parameter area.
 
 // constructors
 

